### PR TITLE
OADP-849: fix replicationDestination race condition

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**

### CODE REVIEW

File: controllers/replicationdestination.go [hunk 1]
```diff
@@ -116,16 +117,24 @@ func (r *VolumeSnapshotRestoreReconciler) SetVSRStatus(log logr.Logger) (bool, e
 	}
 
 	if vsr.Status.Phase == volsnapmoverv1alpha1.SnapMoverRestorePhaseFailed || vsr.Status.Phase == volsnapmoverv1alpha1.SnapMoverRestorePhasePartiallyFailed {
-		return false, errors.New("vsb failed to complete")
+		return false, errors.New("vsr failed to complete")
```
Comment (eemcmullan) [inline controllers/replicationdestination.go:120]: Small request: log vsr name here? Otherwise LGTM.

**Review by eemcmullan — COMMENTED** [review]
Tested changes with app with multiple PVCs and was unable to hit this race condition.

**Comment by eemcmullan** [comment]
Tested changes with app with multiple PVCs and was unable to reproduce race condition. Ty for the fix

File: controllers/replicationdestination.go (hunk 1 above)
Comment (shubham-pampattiwar) [reply controllers/replicationdestination.go]: Done, updated the PR!

**Review by eemcmullan — APPROVED** [review]

**Review by weshayutin — APPROVED** [review]
Thanks @shubham-pampattiwar @eemcmullan

**Comment by shubham-pampattiwar** [comment]
Cherry-pick PR: https://github.com/migtools/volume-snapshot-mover/pull/163

**Review by kaovilai — APPROVED** [review]
Lgtm

**Pull request merged**